third_party/p6/repo/Makefile

```diff
@@ -0,0 +1,227 @@
+################################################################################
```

Comment:
Is this (and other things in this directory) being added directly to the tree instead of as a git submodule?

Reply:
Thanks @bzbarsky-apple  for the review comments. we want to add this directly instead of git submodule. This folder has only *.mtb files for all other dependent assets and modustoolbox software will automatically get all the assets during build process.

Reply:
I guess my concern is that third_party/whatever/repo is typically an actual submodule that's checked out from somewhere.  Other third-party things go into third_party/whatever with other names...

@mspang @andy31415

Reply:
We don't want to add this as submodule and can change the folder from "repo" to something else (p6_sdk) if that is the problem.

Reply:
@bzbarsky-apple : please let me know if I need to change the folder name to p6_sdk (from repo) to address this issue. I can see one more folder in third_party which uses repo folder for files (instead of submodule: third_party/inipp/repo/)

Reply:
I would generally assume "repo" is for submodules.

out of 22 repo.submodules, I see 20 having .git and the only ones not having .git are "inipp" and "lwip" which seem to contain a version and url of a manual (non submodule) checkout.

Reply:
What is the actual buildprocess and what does modustoolbox do?

I have concerns about build  process fetching things: we are hurting in CI times and complexity for things that seem to fetch things. We currently have (and try to clean up):
   - nrfconnect/zephyr does a SDK update at build time (even though mostly not needed)
   - android builds fetch the Android NDK
   - cirque compiles a docker image to run commands in
   
I would prefer a state where all platform SDKs are part of our build environment and if something is really a git submodule, we have it as a clear submodule (at which point we can time checkout duration and decide what to do)

Reply:
As a separate item, can we build with GN instead of makefiles?

Reply:
@andy31415 : Thanks for the review comments.
1. I will change the folder name to p6_sdk as "repo" name is used for submodules (if we plan to support as git submodule in the future, it will be changed to 'repo')
2. What is the actual buildprocess and what does modustoolbox do?
    - Modustoolbox software/Build process
        - Helps to checkout the code from the links mentioned in all *.mtb files in deps folder
        - Then it runs discovery process to find the list of files to be compiled and create JSON file which has all sources required for compile (Refer to "make_gni.py" file)
        - Then p6_sdk.gni updates all file sources from json to gni format.
       - then all the files are compiled using GN
3.  As a separate item, can we build with GN instead of makefiles?
    - Modustoolbox software is not used for building p6_sdk sources, it is used for getting all the assets from github links and getting list of files to be compiled dynamically so that future maintenance is easier.
    - So we are already building with GN.
       
Build time:
1. Getting all assets from GitHub(modustoolbox : make getlibs) as part of build process takes ~ 3mins 
2. Compile time = ~ 6 mins